firedrake/extrusion_utils.py

```diff
@@ -86,9 +97,10 @@ def _get_lp_domains(_inames, _extents):
         domains.extend(_get_lp_domains(('c', 'l'), (base_coord_dim, 2)))
         instructions = """
         ext_coords[{dd}, l, c] = base_coords[{dd}, c]
-        ext_coords[{dd}, l, {base_coord_dim}] = layer_height[0] * (layer + l)
+        ext_coords[{dd}, l, {base_coord_dim}] = layer_height[{hv}] * (layer + l)
```

Comment:
I think the correct formula here now is $(\sum_{i=0}^{layer}layer_height[i]) + layer_height[layer] * l $.

Reply:
Oops, you are completely correct!